src/App.tsx

```diff
@@ -1,23 +1,37 @@
+import SplashScreen from '@pages/Auth/SplashScreen';
 import { useState, useEffect } from 'react';
 import Navigator from '@routes/index';
-import SplashScreen from '@components/SplashScreen';
 import './styles/App.scss';
+import { auth } from '@config/firebase';
+import { onAuthStateChanged } from 'firebase/auth';
+import { useNavigate } from 'react-router-dom';
 
 const App = () => {
+  const navigate = useNavigate();
   const [showSplash, setShowSplash] = useState(true);
 
   useEffect(() => {
     const timer = setTimeout(() => {
```

Comment:
can you pls describe why do you need it?

Reply:
if the user is not logged in, he will be redirected to the login page